src/components/Videos.vue

```diff
@@ -121,6 +139,227 @@ export default {
+					kw: 'Cornish',
+					ky: 'Kirghiz',
+					la: 'Latin',
+					lb: 'Luxembourgish',
+					lg: 'Ganda',
+					li: 'Limburgan',
+					ln: 'Lingala',
+					lo: 'Lao',
+					lt: 'Lithuanian',
+					lu: 'Luba-Katanga',
+					lv: 'Latvian',
+					mg: 'Malagasy',
+					mh: 'Marshallese',
+					mi: 'Maori',
+					mk: 'Macedonian',
+					ml: 'Malayalam',
+					mn: 'Mongolian',
+					mr: 'Marathi',
+					ms: 'Malay',
+					mt: 'Maltese',
+					my: 'Burmese',
+					na: 'Nauru',
+					nb: 'Norwegian Bokmål',
+					nd: 'Ndebele, North',
+					ne: 'Nepali',
+					ng: 'Ndonga',
+					nl: 'Dutch',
+					nn: 'Norwegian Nynorsk',
+					no: 'Norwegian',
+					nr: 'Ndebele, South',
+					nv: 'Navajo',
+					ny: 'Nyanja',
+					oc: 'Occitan',
+					oj: 'Ojibwa',
+					om: 'Oromo',
+					or: 'Oriya',
+					os: 'Ossetian',
+					pa: 'Panjabi',
+					pi: 'Pali',
+					pl: 'Polish',
+					ps: 'Pushto',
+					pt: 'Portuguese',
+					qu: 'Quechua',
+					rm: 'Romansh',
+					rn: 'Rundi',
+					ro: 'Romanian',
+					ru: 'Russian',
+					rw: 'Kinyarwanda',
+					sa: 'Sanskrit',
+					sc: 'Sardinian',
+					sd: 'Sindhi',
+					se: 'Northern Sami',
+					sg: 'Sango',
+					si: 'Sinhala',
+					sk: 'Slovak',
+					sl: 'Slovenian',
+					sm: 'Samoan',
+					sn: 'Shona',
+					so: 'Somali',
+					sq: 'Albanian',
+					sr: 'Serbian',
+					ss: 'Swati',
+					st: 'Sotho',
+					su: 'Sundanese',
+					sv: 'Swedish',
+					sw: 'Swahili',
+					ta: 'Tamil',
+					te: 'Telugu',
+					tg: 'Tajik',
+					th: 'Thai',
+					ti: 'Tigrinya',
+					tk: 'Turkmen',
+					tl: 'Tagalog',
+					tn: 'Tswana',
+					to: 'Tonga',
+					tr: 'Turkish',
+					ts: 'Tsonga',
+					tt: 'Tatar',
+					tw: 'Twi',
+					ty: 'Tahitian',
+					ug: 'Uighur',
+					uk: 'Ukrainian',
+					ur: 'Urdu',
+					uz: 'Uzbek',
+					ve: 'Venda',
+					vi: 'Vietnamese',
+					vo: 'Volapük',
+					wa: 'Walloon',
+					wo: 'Wolof',
+					xh: 'Xhosa',
+					yi: 'Yiddish',
+					yo: 'Yoruba',
+					za: 'Zhuang',
+					zh: 'Chinese',
+					zu: 'Zulu',
+				}
+				const davDir = this.davPath.replace(/[^/]*$/, '')
+				const videoRoot = this.basename.replace(/[.][^.]+$/, '')
+				// Create caption tracks for the HTML5 player
+				// E.g.: <file>.mkv: look for <file>.xx.vtt or .<file>.xx.vtt
+				const capTracks = []
+				for (const file of folder) {
+					const basename = file.basename
+					const index = basename.indexOf(videoRoot)
+					// Consider only file... or .file...
+					if (!(index === 0 || (index === 1 && basename[0] === '.'))) {
+						continue
+					}
+					const suffix = basename.slice(videoRoot.length + index)
+					// Consider only ...xx.vtt
+					if (suffix.search(/^[.]..[.]vtt$/) !== 0) {
+						continue
+					}
+					const lang = suffix.slice(1, 3)
+					const language = languages[lang] || lang
+					// an array of objects with src, label and lang keys
+					capTracks.push({
+						src: davDir + basename,
+						label: language,
+						srclang: lang,
+					})
+				}
+				this.tracks = capTracks
```

Comment:
I'm wondering if that should not be better to have all of this in a dedicated php Controller that, given a specific fileName, that would return an array of `LANG => filename`.

I think the code would benefit from it, especially because the Locale list is managed by a php method.

Reply:
Not sure how to do that (writing a dedicated php Controller).
Please would you point to a similar Controller in NextCloud source, which I could use as a template?

Reply:
I think the controller method should:

1. Receive the filename as a parameter
2. Search for existing files with the same filename+langcode
3. Return an array of `langcode => filename` so the viewer can use it and build the list accordingly. Should be much faster than our dav endpoint.

Reply:
OK. I take the challenge :)

By the way, the current proposal does not use the dav endpoint to look for candidate vtt files -- I reckon getFileList() works with the local filesystem?
So we may not see a performance improvement with your proposal.
But in any case the code will be cleaner and easier to maintain.